common/buffer.c

```diff
@@ -77,8 +77,14 @@ bool
 p11_buffer_init_null (p11_buffer *buffer,
                       size_t reserve)
 {
+	bool ok;
+
+	return_val_if_fail (reserve < SIZE_MAX, false);
+
 	p11_buffer_init_full (buffer, NULL, 0, P11_BUFFER_NULL, realloc, free);
-	return buffer_realloc (buffer, reserve);
+	ok = buffer_realloc (buffer, reserve + 1);
+	((unsigned char *)buffer->data)[reserve] = '\0';
```

Comment:
I dont think this is an intuitive behavior. This probably deserves a function documentation comment.

Reply:
In the end I reverted this change and switched to use `p11_buffer_init_null` followed by `p11_buffer_append` (which does append NUL byte as necessary).